activestorage/lib/active_storage/downloader.rb

```diff
@@ -35,8 +35,9 @@ def download(key, file)
       end
 
       def verify_integrity_of(file, checksum:)
-        unless ActiveStorage.checksum_implementation.file(file).base64digest == checksum
-          raise ActiveStorage::IntegrityError
+        actual_checksum = service.compute_checksum(file)
```

Comment:
This utility file is weird to have: it is only used in one place and only has one public method. How about merging it directly into service to make sure it is consistently overwritten if needed?